src/ifcparse/IfcSpfHeader.cpp

```diff
@@ -36,19 +36,22 @@ static const char * const DATA				= "DATA";
 using namespace IfcParse;
 
 void IfcSpfHeader::readSemicolon() {
-	if (!TokenFunc::isOperator(_lexer->Next(), ';')) {
+    Token next = _lexer->Next();
+    if (next.lexer != 0 && !TokenFunc::isOperator(next, ';')) {
```

Comment:
I did not look into this really in-depth as noticed the `IfcSpfLexer* lexer; //TODO: remove it from here` comment. Can be refactored properly when time permits.